runtime/rastrace/trcengine.c

```diff
@@ -333,6 +334,7 @@ J9VMDllMain(J9JavaVM *vm, IDATA stage, void *reserved)
 		 * Find UTE thread slot inside VM thread structure
 		 */
 		tempThr = UT_THREAD_FROM_VM_THREAD(vm->mainThread);
+		*tempThr = &thrData;
```

Comment:
I think some comments (or assertions) would be helpful:
* before line 337 `*tempThr` is expected to be `NULL`
* after the call to `initializeTrace()` `*tempThr` is again expected to be `NULL` (eliminating the reference to `thrData`)
* the call to `threadStart()` should allocate storage to which `*tempThr` will then point

Reply:
> the call to `threadStart()` should allocate storage to which `*tempThr` will then point

`threadStart()` does the same as `initializeTrace()`: Both of them take `UtThreadData **thr` as one of their arguments, and assign the address of a local variable to `*thr`.  GCC 13.2.0 complains only about `initializeTrace()`.  Why?
https://github.com/eclipse-openj9/openj9/blob/c5ef76214cd1f89bc41ebee23df4b9e8eeab9547/runtime/rastrace/trcmain.c#L992

Reply:
Now I understand.
There is a path in `initializeTrace()` which does not clear the reference to the local variable in `*thr`, and that caused the compilation error.  Adding `*thr = NULL;` in the path resolved the error.
I am updating the commit.